Tests/Tracking/CustomerIOTest.swift

```diff
@@ -15,19 +13,20 @@ class CustomerIOTest: UnitTest {
     override func setUp() {
         super.setUp()
 
-        diGraph.override(value: hooksMock, forType: HooksManager.self)
-        diGraph.override(value: cleanupRepositoryMock, forType: CleanupRepository.self)
         diGraph.override(value: globalDataStoreMock, forType: GlobalDataStore.self)
 
         customerIO = CustomerIO(implementation: implmentationMock, diGraph: diGraph)
     }
 
-    func test_initialize_expectAddModuleHooks_expectRunCleanup() {
-        customerIO.postInitialize(diGraph: diGraph, module: TrackingModuleHookProvider(), cleanupRepositoryImp: cleanupRepositoryMock)
-
-        XCTAssertEqual(hooksMock.addCallsCount, 1)
-        XCTAssertEqual(hooksMock.addReceivedArguments?.key, .tracking)
+    func test_initialize_givenPushDeviceTokenNotSet_expectRegisterDeviceTokenNotCalled() {
+        customerIO.postInitialize(diGraph: diGraph)
+        XCTAssertFalse(implmentationMock.registerDeviceTokenCalled)
+    }
 
-        XCTAssertEqual(cleanupRepositoryMock.cleanupCallsCount, 1)
+    func test_initialize_givenPushDeviceTokenSet_expectRegisterDeviceTokenCalled() {
+        let pushDeviceToken = String.random
+        globalDataStoreMock.pushDeviceToken = pushDeviceToken
+        customerIO.postInitialize(diGraph: diGraph)
+        XCTAssertTrue(implmentationMock.registerDeviceTokenCalled)
     }
```

Comment:
Looking at [the customerio class](https://github.com/customerio/customerio-ios/blob/shahroz/track-cleanup/Sources/Tracking/CustomerIO.swift), I don't see anything in that code about device tokens. 

Am I looking at the wrong class? I don't understand why there are device token tests written in this PR.

Reply:
Please [look into this class](https://github.com/customerio/customerio-ios/blob/shahroz/track-cleanup/Sources/Common/CustomerIOInstance.swift) for the correct method call.

Reply:
Oh, ok. 

The test file that was modified is `Tests/Tracking/CustomerIOTest.swift`. The [class you linked to](https://github.com/customerio/customerio-ios/blob/shahroz/track-cleanup/Sources/Common/CustomerIOInstance.swift?rgh-link-date=2024-01-09T17%3A50%3A29Z) is `Sources/Common/CustomerIOInstance.swift`. 

I was expecting that the file in the `Tests/` directory map to the same file in the `Sources/` directory. 

I can understand why this tests file was modified to remove the module hooks logic. However, I would expect that the newly added test functions be moved into a different file.

Reply:
After carefully looking at the difference between what you suggest and what has been done currently, I think it does make a lot value to bring this class to `Tests/Common`. The other method in this class also calls the same function and [links to this class](https://github.com/customerio/customerio-ios/blob/shahroz/track-cleanup/Sources/Common/CustomerIOInstance.swift?rgh-link-date=2024-01-09T17%3A50%3A29Z).

Thanks for pointing out.

Reply:
Find the change in the [latest commit](https://github.com/customerio/customerio-ios/pull/471/commits/40e31c4c213116ef3dc915e99d8a4b9f410359ef).